src/Widgets/FeedRow.vala

```diff
@@ -152,19 +156,28 @@ public class FeedReader.FeedRow : Gtk.ListBoxRow {
 	{
 		var icon = new Gtk.Image.from_icon_name("feed-rss-symbolic", Gtk.IconSize.LARGE_TOOLBAR);
 		icon.get_style_context().add_class("fr-sidebar-symbolic");
-
-		var manager = FavIconManager.get_default();
-		manager.getIcon.begin(m_feed, (obj, res) => {
-			var pixbuf = manager.getIcon.end(res);
-			if(pixbuf != null)
-			{
-				icon.pixbuf = pixbuf;
-				icon.get_style_context().remove_class("fr-sidebar-symbolic");
-			}
-		});
+		reloadFavIcon.begin(icon);
 		return icon;
 	}
 
+	public async void reloadFavIcon(Gtk.Image? inIcon = null)
+	{
+		var icon = yield FavIconManager.get_default().getIcon(m_feed);
+		if(icon == null)
+			return;
+
+		if(inIcon == null)
+		{
+			m_icon.pixbuf = icon;
+			m_icon.get_style_context().remove_class("fr-sidebar-symbolic");
+		}
+		else
+		{
+			inIcon.pixbuf = icon;
+			inIcon.get_style_context().remove_class("fr-sidebar-symbolic");
+		}
```

Comment:
Same thing as above: This condition seems way more complicated than just always passing in the icon.